diff --git a/arch/arm/boot/dts/bcm2837.dtsi b/arch/arm/boot/dts/bcm2837.dtsi
index 5dbdebc46259..3357f7446a54 100644
--- a/arch/arm/boot/dts/bcm2837.dtsi
+++ b/arch/arm/boot/dts/bcm2837.dtsi
@@ -38,7 +38,6 @@ timer {
 	cpus: cpus {
 		#address-cells = <1>;
 		#size-cells = <0>;
-		enable-method = "brcm,bcm2836-smp"; // for ARM 32-bit
 
 		/* Source for d/i-cache-line-size and d/i-cache-sets
 		 * https://developer.arm.com/documentation/ddi0500/e/level-1-memory-system
@@ -51,8 +50,7 @@ cpu0: cpu@0 {
 			device_type = "cpu";
 			compatible = "arm,cortex-a53";
 			reg = <0>;
-			enable-method = "spin-table";
-			cpu-release-addr = <0x0 0x000000d8>;
+			enable-method = "psci";
 			d-cache-size = <0x8000>;
 			d-cache-line-size = <64>;
 			d-cache-sets = <128>; // 32KiB(size)/64(line-size)=512ways/4-way set
@@ -66,8 +64,7 @@ cpu1: cpu@1 {
 			device_type = "cpu";
 			compatible = "arm,cortex-a53";
 			reg = <1>;
-			enable-method = "spin-table";
-			cpu-release-addr = <0x0 0x000000e0>;
+			enable-method = "psci";
 			d-cache-size = <0x8000>;
 			d-cache-line-size = <64>;
 			d-cache-sets = <128>; // 32KiB(size)/64(line-size)=512ways/4-way set
@@ -81,8 +78,7 @@ cpu2: cpu@2 {
 			device_type = "cpu";
 			compatible = "arm,cortex-a53";
 			reg = <2>;
-			enable-method = "spin-table";
-			cpu-release-addr = <0x0 0x000000e8>;
+			enable-method = "psci";
 			d-cache-size = <0x8000>;
 			d-cache-line-size = <64>;
 			d-cache-sets = <128>; // 32KiB(size)/64(line-size)=512ways/4-way set
@@ -96,8 +92,7 @@ cpu3: cpu@3 {
 			device_type = "cpu";
 			compatible = "arm,cortex-a53";
 			reg = <3>;
-			enable-method = "spin-table";
-			cpu-release-addr = <0x0 0x000000f0>;
+			enable-method = "psci";
 			d-cache-size = <0x8000>;
 			d-cache-line-size = <64>;
 			d-cache-sets = <128>; // 32KiB(size)/64(line-size)=512ways/4-way set
@@ -121,6 +116,10 @@ l2: l2-cache0 {
 			cache-level = <2>;
 		};
 	};
+		psci {
+		compatible = "arm,psci-1.0", "arm,psci-0.2";
+		method = "smc";
+	};
 };
 
 /* Make the BCM2835-style global interrupt controller be a child of the
diff --git a/arch/arm/configs/bcmrpi_defconfig b/arch/arm/configs/bcmrpi_defconfig
index 83a11f65368a..d47f37b5ec25 100644
--- a/arch/arm/configs/bcmrpi_defconfig
+++ b/arch/arm/configs/bcmrpi_defconfig
@@ -30,14 +30,15 @@ CONFIG_BLK_DEV_INITRD=y
 CONFIG_EMBEDDED=y
 # CONFIG_COMPAT_BRK is not set
 CONFIG_PROFILING=y
-CONFIG_ARCH_MULTI_V6=y
-# CONFIG_ARCH_MULTI_V7 is not set
+CONFIG_ARCH_MULTI_V7=y
 CONFIG_ARCH_BCM=y
 CONFIG_ARCH_BCM2835=y
 # CONFIG_CACHE_L2X0 is not set
 # CONFIG_CPU_SW_DOMAIN_PAN is not set
 CONFIG_UACCESS_WITH_MEMCPY=y
 # CONFIG_ATAGS is not set
+CONFIG_ARM_PSCI=y
+CONFIG_HAVE_ARM_SCMCCC=y
 CONFIG_CMDLINE="console=ttyAMA0,115200 kgdboc=ttyAMA0,115200 root=/dev/mmcblk0p2 rootfstype=ext4 rootwait"
 CONFIG_CPU_FREQ=y
 CONFIG_CPU_FREQ_STAT=y
